Extension/src/common.ts

```diff
@@ -1304,3 +1304,21 @@ export function sequentialResolve<T>(items: T[], promiseBuilder: (item: T) => Pr
         return promiseBuilder(nextItem);
     }, Promise.resolve());
 }
+
+export function normalizeArg(arg: string): string {
+    arg = arg.trimLeft().trimRight();
+    // Check if the arg is enclosed in backtick,
+    // or includes escaped double-quotes,
```

Comment:
This is a really hard problem, actually...  I think we really need a test case for this, so we can plug in various different scenarios to ensure they are supported. 

Also, contrary to what I may have said (wrong) in a previous conversation, I think you want to check for any "unescaped" double-quotes here.  We should ignore any arg with un-escaped quotes in it, as adding quotes around that (even if escaping them internally) changes the arg...  For example:

```
$ g++ -E -dD -xc++ -DTESTDEFINE="asd asd" /dev/null | grep TESTDEFINE
#define TESTDEFINE asd asd
```

would become:

```
$ g++ -E -dD -xc++ "-DTESTDEFINE=\"asd asd\"" /dev/null | grep TESTDEFINE
#define TESTDEFINE "asd asd"
```